server/src/main/scala/org/scassandra/server/cqlmessages/CqlProtocolHelper.scala

```diff
@@ -59,99 +79,87 @@ object CqlProtocolHelper {
     frameBuilder.result().toArray
   }
 
-  def serializeBooleanValue(bool : Boolean) : Array[Byte] = {
-    Array[Byte](0,0,0,1, if (bool) 1 else 0)
-  }
-
-  def serializeBigIntValue(value : Long) : Array[Byte] = {
+  def serializeCollection[U](sizeF: (ByteStringBuilder, Int) => ByteStringBuilder)(input: TraversableOnce[U], entrySerializer: U => Array[Byte]) : Array[Byte] = {
     val frameBuilder = ByteString.newBuilder
-    frameBuilder.putInt(8)
-    frameBuilder.putLong(value)
-    frameBuilder.result().toArray
-  }
-
-  def serializeDecimalValue(decimal: java.math.BigDecimal) : Array[Byte] = {
-    val scale = decimal.scale()
-    val unscaledValue = decimal.unscaledValue().toByteArray
-    val length = 4 + unscaledValue.length
-    val frameBuilder = ByteString.newBuilder
-    frameBuilder.putInt(length)
-    frameBuilder.putInt(scale)
-    frameBuilder.putBytes(unscaledValue)
+    sizeF(frameBuilder, input.size)
+    for (value <- input) {
+      frameBuilder.putBytes(entrySerializer(value))
+    }
     frameBuilder.result().toArray
   }
 
-  def serializeDoubleValue(doubleValue : Double) : Array[Byte] = {
-    val frameBuilder = ByteString.newBuilder
-    frameBuilder.putInt(8)
-    frameBuilder.putDouble(doubleValue)
-    frameBuilder.result().toArray
+  def  serializeCqlCollection[T,U](input: TraversableOnce[U], entrySerializer: Any => Array[Byte])(implicit protocolVersion: ProtocolVersion) = {
+    // protocol <= v2 use short length sizing, >= v3 uses int.
+    val serializer: (TraversableOnce[U], U => Array[Byte]) => Array[Byte] = protocolVersion match {
+      case VersionOne | VersionTwo => serializeCollection(_.putShort(_))
+      case VersionThree => serializeCollection(_.putInt(_))
+    }
+    serializer(input, entrySerializer)
```

Comment:
Maybe we could have this as a class rather than a object and have different implementations for the various versions. Would then be very explicit when they differ. Like with the CqlMessagefactory